pkg/kubectl/cmd/create.go

```diff
@@ -343,7 +343,11 @@ func RunCreateSubcommand(f cmdutil.Factory, cmd *cobra.Command, out io.Writer, o
 	obj = info.Object
 
 	if !options.DryRun {
-		obj, err = resource.NewHelper(client, mapping).Create(namespace, false, info.Object)
+		help := resource.NewHelper(client, mapping)
+		if !help.NamespaceScoped && nsOverriden {
+			return fmt.Errorf("%s is cluster-scope resources, can not special --namespace %s", gvk.Kind, namespace)
```

Comment:
> %s is cluster-scope resources, can not special --namespace %s

unfortunately this is not a good error message grammatically.

Please consider:

> %s is not a namespaced resource, --namespace should not be specified